File: apps/monolith/src/main/java/com/example/iter/admin/payment/dto/AdminPaymentSummaryResponse.java
package com.example.iter.admin.payment.dto;

import com.example.iter.payment.api.PaymentStatus;
import com.example.iter.reservation.api.RentalStatus;

import java.math.BigDecimal;
import java.time.LocalDateTime;

public record AdminPaymentSummaryResponse(
        Long paymentId,
        Long rentalId,
        String orderId,
        Long renterId,
        String renterEmail,
        String renterName,
        String renterNickname,
        String equipmentName,
        BigDecimal amount,
        PaymentStatus paymentStatus,
        RentalStatus rentalStatus,
        LocalDateTime paidAt,
        LocalDateTime refundedAt,
        LocalDateTime createdAt
) {
}

File: apps/monolith/src/main/kotlin/iter/admin/payment/dto/AdminPaymentSummaryResponse.kt
package iter.admin.payment.dto

import iter.payment.api.PaymentStatus
import iter.reservation.api.RentalStatus
import java.math.BigDecimal
import java.time.LocalDateTime
import kotlin.jvm.JvmRecord

@JvmRecord
data class AdminPaymentSummaryResponse(
    val paymentId: Long?,
    val rentalId: Long?,
    val orderId: String?,
    val renterId: Long?,
    val renterEmail: String?,
    val renterName: String?,
    val renterNickname: String?,
    val equipmentName: String?,
    val amount: BigDecimal?,
    val paymentStatus: PaymentStatus?,
    val rentalStatus: RentalStatus?,
    val paidAt: LocalDateTime?,
    val refundedAt: LocalDateTime?,
    val createdAt: LocalDateTime?
)
